files/nrpe/check_openstack.py

```diff
@@ -286,23 +287,42 @@ def instance_destroy(self):
       self.nova.servers.delete(self.instance.id)
 
   def instance_attach_floating_ip(self):
-    self.fip = self.nova.floating_ips.create(self.options.fip_pool)
-    self.instance.add_floating_ip(self.fip)
+    # On admin-user, this is problematic since it can see multiple 'public' nets
+    # On non-admin-user, no issue as long as a net called 'public' isn't created to the project
+    public_network_id = self.neutron.list_networks(
+                         name='public')['networks'][0]['id']
+    body = {'floatingip': {'floating_network_id': public_network_id}}
+    self.fip = self.neutron.create_floatingip(body)
+    # this bit courtesy of https://hammers.readthedocs.io/en/latest/_modules/ccmanage/server.html
+    try:
+        self.instance.add_floating_ip(self.fip['floatingip']['floating_ip_address'])
+    except AttributeError:
+        ports = self.neutron.list_ports(**{'device_id': self.instance.id}).get('ports')
+        fip_target = {
+            'port_id': ports[0]['id'],
+            'ip_addr': ports[0]['fixed_ips'][0]['ip_address']
+        }
+        target_id = fip_target['port_id']
+        self.neutron.update_floatingip(self.fip['floatingip']['id'], body={
+            'floatingip': {
+                'port_id': target_id,
+            }
+        })
 
   def instance_detach_floating_ip(self):
     if hasattr(self, 'fip'):
       self.instance.remove_floating_ip(self.fip.ip)
-      self.nova.floating_ips.delete(self.fip)
+      self.neutron.delete_floatingip(self.fip['floatingip']['id'])
 
   def floating_ip_delete(self):
     if hasattr(self, 'fip'):
-      self.nova.floating_ips.delete(self.fip)
+      self.neutron.delete_floatingip(self.fip['floatingip']['id'])
```

Comment:
From video review with team: This should be checked that it doesn't do anything dangerous if run as admin.

Reply:
See above